language/move-lang/tests/functional/compare/set.move

```diff
@@ -0,0 +1,118 @@
+// Generic set that leverages Compare::cmp.
+// This is a reasonable smoke test for the Compare module, but don't actually use this without
+// singificantly more testing/thought about the API!
+module Set {
+    use 0x0::Compare;
+    use 0x0::Transaction;
+    use 0x0::Vector;
+
+    struct T<Elem> { v: vector<Elem> }
+
+    public fun empty<Elem>(): Self::T<Elem> {
+        T { v: Vector::empty() }
+    }
+
+    public fun size<Elem>(t: &T<Elem>): u64 {
+       Vector::length(&t.v)
+    }
+
+    public fun borrow<Elem>(t: &T<Elem>, index: u64): &Elem {
+        Vector::borrow(&t.v, index)
+    }
```

Comment:
This seems like a really weird API for a `Set` 😝, but I like the test 👍

Reply:
It's definitely weird, but I'm not sure how else to support iteration (which I need for this test)...